docs/reference/readme.md

```diff
@@ -259,6 +259,17 @@ If `false`
 
 Forward authentication creates an endpoint that can be used with third-party proxies that do not have rich access control capabilities ([nginx](http://nginx.org/en/docs/http/ngx_http_auth_request_module.html), [nginx-ingress](https://kubernetes.github.io/ingress-nginx/examples/auth/oauth-external-auth/), [ambassador](https://www.getambassador.io/reference/services/auth-service/), [traefik](https://docs.traefik.io/middlewares/forwardauth/)). Forward authentication allows you to delegate authentication and authorization for each request to Pomerium.
 
+### Forward Auth Type
```

Comment:
+ Upgrading guide that this is now a required field.

Reply:
I'm leaning to make docs change in other PR. Also thinking of should we always mandate the forward auth service to be run along side others? I'm afraid user will be frustrated because they don't use forward auth but still have to config it to make pomerium run.

Reply:
We should only require the type if they configure a forward auth url, I think.

We should definitely put due consideration into upgrade pain on this one.

Reply:
@travisgroth Added that constraint, only start forward auth service when forward auth url not empty

Reply:
> We should definitely put due consideration into upgrade pain on this one.

I really really agree with that. Let's discuss before merging.